pkg/sql/main_test.go

```diff
@@ -155,12 +155,8 @@ func checkEndTransactionTrigger(args storagebase.FilterArgs) *roachpb.Error {
 
 	var hasSystemKey bool
 	for _, span := range req.IntentSpans {
-		keyAddr, err := keys.Addr(span.Key)
-		if err != nil {
-			return roachpb.NewError(err)
-		}
-		if bytes.Compare(keyAddr, keys.SystemConfigSpan.Key) >= 0 &&
-			bytes.Compare(keyAddr, keys.SystemConfigSpan.EndKey) < 0 {
+		if bytes.Compare(span.Key, keys.SystemConfigSpan.Key) >= 0 &&
```

Comment:
We want to set the system config trigger for any *data* records in the system config span. We can determine that by looking at the keys alone. Using keys.Addr finds any records that live on the same range as the system config span, including range descriptors and transaction records. This bug was revealed when the split points changed so that range descriptors fall into this range.